webrtc.html

```diff
@@ -3088,25 +3101,96 @@ <h2>Methods</h2>
-                        slot.</p>
-                      </li>
                       <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">
-                        <p>If <var>description.sdp</var> is the empty string and
-                        <var>description.type</var> is <code>"offer"</code>,
-                        then set <var>description.sdp</var> to the value of
+                        <p>If <var>type</var> is <code>"offer"</code>, and
+                        <var>sdp</var> is not the empty string and not equal to
                         <var>connection</var>'s <a>[[\LastCreatedOffer]]</a>
-                        slot.</p>
+                        slot, then return a promise <a>rejected</a> with a newly
+                        <a data-link-for="exception" data-lt="create">created</a>
+                        <code>InvalidModificationError</code> and abort these
+                        steps.</p>
+                      </li>
+                      <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">
+                        <p>If <var>type</var> is <code>"answer"</code> or
+                        <code>"pranswer"</code>, and <var>sdp</var> is not the
+                        empty string and not equal to <var>connection</var>'s
+                        <a>[[\LastCreatedAnswer]]</a> slot, then return a
+                        promise <a>rejected</a> with a newly
+                        <a data-link-for="exception" data-lt="create">created</a>
+                        <code>InvalidModificationError</code> and abort these
+                        steps.</p>
+                      </li>
+                      <li>
+                        <p>If <var>sdp</var> is the empty string, and
+                          <var>type</var> is <code>"offer"</code>, then run the
+                          following sub steps:</p>
+                        <ol>
+                          <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">
+                            <p>Set <var>sdp</var> to the value of
+                            <var>connection</var>'s <a>[[\LastCreatedOffer]]</a>
+                            slot.</p>
+                          </li>
+                          <li data-tests="RTCPeerConnection-setLocalDescription-offer.html">
+                            <p>If <var>sdp</var> is the empty string, or if it
+                            no longer accurately represents the system state of
+                            <var>connection</var>, or if <var>connection</var>'s
+                            configured indentity provider has changed since
+                            <var>connection</var>'s <a>[[\LastCreatedOffer]]</a>
+                            was created, then let <var>p</var> be the result of
+                            <a href="#dfn-create-an-offer">creating an offer</a>
+                            with <var>connection</var>, and return the result of
```

Comment:
Can you verify that the following is true?
Because these steps are chained in the connection's operation queue, there is no risk that the offer created here becomes invalid by the time that SLD() is performed. Thus no validation is required of the result of creating the offer.

Reply:
No, the operation chain isolates us from the other negotiation methods that also use the chain—*createOffer*, *createAnswer*, *setLocalDescription*, *setRemoteDescription*, *addIceCandidate* and *replaceTrack*—but not from other modification methods that *require* negotiation like e.g. *addTransceiver*, *addTrack*, *createDataChannel*, *setStreams*.

This PR text mirrors existing text in the [final steps to create an offer](https://w3c.github.io/webrtc-pc/#dfn-final-steps-to-create-an-offer) (in fact I almost linked to it, but it read a bit odd and didn't work right).